packages/common/src/createNonceManager.ts

```diff
@@ -75,11 +75,8 @@ export function createNonceManager({
     await ref.noncePromise;
   }
 
-  function shouldResetNonce(error: unknown): boolean {
-    return (
-      error instanceof BaseError &&
-      error.walk((e) => e instanceof NonceTooLowError || e instanceof NonceTooHighError) != null
-    );
+  function shouldResetNonce(error: Error): boolean {
+    return error.name === "NonceTooLowError" || error.name === "NonceTooHighError";
```

Comment:
did you test this? I think we still need to use `BaseError` and `error.walk` because these nonce errors are deeply nested

Reply:
ahh right yes, need to `walk`. Updated!

Reply:
Tested locally by setting nonce + 1 and it works now